libyul/optimiser/Suite.cpp

```diff
@@ -267,44 +267,29 @@ map<char, string> const& OptimiserSuite::stepAbbreviationToNameMap()
 	return lookupTable;
 }
 
+
+
 void OptimiserSuite::validateSequence(string const& _stepAbbreviations)
 {
+	stack<char> s;
+   	char ch;
 	bool insideLoop = false;
-	for (char abbreviation: _stepAbbreviations)
-		switch (abbreviation)
-		{
-		case ' ':
-		case '\n':
-			break;
-		case '[':
-			assertThrow(!insideLoop, OptimizerException, "Nested brackets are not supported");
-			insideLoop = true;
-			break;
-		case ']':
-			assertThrow(insideLoop, OptimizerException, "Unbalanced brackets");
-			insideLoop = false;
-			break;
-		default:
-		{
-			yulAssert(
-				string(NonStepAbbreviations).find(abbreviation) == string::npos,
-				"Unhandled syntactic element in the abbreviation sequence"
-			);
-			assertThrow(
-				stepAbbreviationToNameMap().find(abbreviation) != stepAbbreviationToNameMap().end(),
-				OptimizerException,
-				"'"s + abbreviation + "' is not a valid step abbreviation"
-			);
-			optional<string> invalid = allSteps().at(stepAbbreviationToNameMap().at(abbreviation))->invalidInCurrentEnvironment();
-			assertThrow(
-				!invalid.has_value(),
-				OptimizerException,
-				"'"s + abbreviation + "' is invalid in the current environment: " + *invalid
-			);
-
-		}
-		}
-	assertThrow(!insideLoop, OptimizerException, "Unbalanced brackets");
+	int stepAbb_length = _stepAbbreviations.size();
+   for (int i=0; i<stepAbb_length; i++) 
```

Comment:
Why not use a range loop here? You only use `i` for `_stepAbbreviations[i]` so you could easily replace it with a variable representing the current character and a range loop is exactly that. This is why this code was originally using one:
```suggestion
   for (char abbreviation: _stepAbbreviations)
```

Reply:
Thanks for the feedback @cameel i will refer and modify the commits accordingly